### ISSUE TITLE
Disable main window

### ISSUE THREAD

**Issue by Zettexe** [98teg/NativeDialogs#7]
How would I disable interactions on the main game window while a dialog is open? I can't seem to find a way to do it with Godot.

**Comment by 98teg** [comment]
The best solution is to create a control node that occupies the entire screen, setting its mouse filter flag to `stop`. Then, make this node visible whenever you want the user to not interact with the UI, and hide it again to give control back to the user. I found this trick [here](https://godotengine.org/qa/14354/can-i-freeze-a-part-of-my-gui)

**Comment by Zettexe** [comment]
That would work, however I am looking for something like the Windows Notepad Save File as dialogue where it completely blocks access to the main window until the save dialog is finished.

**Comment by 98teg** [comment]
Not sure if that is possible, sorry

**Comment by Zettexe** [comment]
I looked into other solutions and Godot 4's new Window class with the Exclusive and Popup flags set actually works exactly how I would want it. 
Is there any possibility to include some if not all of the new Window class flags into this project for Godot 4? 
I can manage all this from my own game of course but would be nice if it was built in.

**Comment by Zettexe** [comment]
My current solution is to simply open a new Window 1x1px in size at position x: -1, y: -1 making it invisible and disabling all rendering related things associated to it. Its not a perfect solution since it doesn't force the dialogs to the top when you click the main window but hopefully I can find a solution to that soon, might be some way to associate the dialogs to the new window.

**Comment by 98teg** [comment]
The only thing I can think of is to detect the user focusing on your program's main window to call the show method again. The library I use for this plugin doesn't have the option to make the windows it creates to be always on top, unfortunately.

**Comment by 98teg** [comment]
Here is how you may achieve that:
```gdscript
var window: Window = get_viewport()
window.connect("focus_entered", show)
```

**Comment by Zettexe** [comment]
Tried it and it doesn't really work as intended, it just ends up creating an infinite loop of reopening the dialog if you close it. 

I did however find something else, create a sub-window with the properties "transient, exclusive and popup_window" set it never forces the Godot window above any other application except for tabbing into it, including the dialogs. Effectively this completely disables any input into the main game window as long as the popup is open.

One caveat though is that the option embed_subwindows must be disabled on the parent window for the OS controls (only tested on Window btw) to be disabled on the window, otherwise it simply disables any standard Godot inputs. Not sure if there would be a better solution with a different plugin perhaps, this seems to be about as much as I can do with the current setup.

**Comment by Zettexe** [comment]
From what I saw in the Popup class, they disable some of the window properties so I imagine you would be able to do so with any property. 

Also I think in this case the embedded subwindows option on the parent is more negative than positive since it completely disables OS controls, including dragging the window around, resizing, and using the OS buttons. Leaving it true (default) only prevents normal Godot inputs. 

I do think that there may be a bit too many inconsistencies with implementing such a thing though, say someone did change embedded subwindows to false then this addons functionality would change along with it and there is also a potential performance draw from just inheriting from Window in the first place.

**Comment by 98teg** [comment]
Yeah, that's why I think documenting this workaround is the best option. Just inform people about this solution and let them judge if it fits their use case.
